### ISSUE TITLE
Interoperability issue between Spring Boot 3 tracing and X-Ray

### ISSUE THREAD

**Issue by dalbani** [openzipkin/zipkin-aws#198]
## Describe the Bug
When combined with Spring Boot 3 tracing functionality, the traces sent to X-Ray are entirely correct.

## Steps to Reproduce
1. Enable tracing in a Spring Boot 3 app with Brave + X-Ray backend, e.g.:
```xml
        <dependency>
            <groupId>io.micrometer</groupId>
            <artifactId>micrometer-tracing-bridge-brave</artifactId>
        </dependency>

        <dependency>
            <groupId>io.zipkin.reporter2</groupId>
            <artifactId>zipkin-reporter-brave</artifactId>
        </dependency>

        <dependency>
            <groupId>io.zipkin.aws</groupId>
            <artifactId>zipkin-reporter-xray-udp</artifactId>
            <version>0.23.4</version>
        </dependency>
```

2. Start the local X-Ray daemon.
3. Send an HTTP request to a (dummy) controller.
4. Check via the AWS console what trace was sent to X-Ray.
<img width="1196" alt="image" src="https://user-images.githubusercontent.com/1470270/215741443-9ed1831b-d7c1-42a2-ab74-fd7f9d5348e0.png">

```json
{
    "Id": "1-63d8f320-fbda6c1430c73761c0faec71",
    "Duration": 0.016,
    "LimitExceeded": false,
    "Segments": [
        {
            "Id": "30c73761c0faec71",
            "Document": {
                "id": "30c73761c0faec71",
                "name": "ticketia",
                "start_time": 1675162400.421386,
                "trace_id": "1-63d8f320-fbda6c1430c73761c0faec71",
                "end_time": 1675162400.436936,
                "http": {
                    "request": {
                        "url": "/ticketia/swagger-ui/swagger-ui-standalone-preset.js",
                        "method": "HTTP GET /SWAGGER-UI*/**"
                    }
                },
                "aws": {
                    "xray": {
                        "sdk": "Zipkin Brave"
                    }
                },
                "annotations": {
                    "exception": "none",
                    "method": "GET",
                    "uri": "/swagger-ui*/**",
                    "outcome": "SUCCESS",
                    "status": "200"
                }
            }
        }
    ]
}
```

## Expected Behaviour

The `method` property should contain `GET` instead of `HTTP GET /SWAGGER-UI*/**`.

-----

After doing a little of debugging, my understanding is that mismatch comes from some naming differences between Spring and Zipkin.
For example, those are the tags of the related span (breakpoint at `UDPMessageEncoder.java#L123` mentioned below):
<img width="352" alt="image" src="https://user-images.githubusercontent.com/1470270/215742396-8f9492cd-8b62-4157-9da6-c1b1bab8ff88.png">
Which are, if I understand correctly, using the names defined in https://github.com/spring-projects/spring-framework/blob/main/spring-web/src/main/java/org/springframework/http/client/observation/ClientHttpObservationDocumentation.java.
That would explain why there's a tag called `http.url` compared to simply `method` and `status` for example.

On the other hand, as per the code at https://github.com/openzipkin/zipkin-aws/blob/master/storage/xray-udp/src/main/java/zipkin2/storage/xray_udp/UDPMessageEncoder.java#L123, the X-Ray UDP encoder expects the following tags:
- `http.method`
- `http.url`
- `http.status_code`

If not present, the code below in this class, at line 221, uses the span name as HTTP method (sic!).

**Bottom line**: does anyone how to fix that? Be it at the Zipkin or Spring Boot side.
Thanks!

**Comment by shakuzen** [comment]
Maybe the default tagging should change, and/or the xray encoder here could be updated to handle this case, but in the meantime, an option you have is to customize the tags that Spring instrumentation uses by providing your own `ClientRequestObservationConvention`. You could extend the [DefaultClientRequestObservationConvention](https://github.com/spring-projects/spring-framework/blob/main/spring-web/src/main/java/org/springframework/http/client/observation/DefaultClientRequestObservationConvention.java). You could change the tag for the HTTP method to `http.method` as expected in the encoder, or you could double tag with the current `method` and a duplicate `http.method` tag. Would you try that and let us know if that resolves the issue?

**Comment by codefromthecrypt** [comment]
possibly the update to the README here helps, and something could detect this config and add it by default? #197

**Comment by codefromthecrypt** [comment]
I feel like we ran into this in sleuth, but basically right now the only hint that x-ray is in use is if propagation is the AWS one.

If that's present make sure the data collection policy matches x-ray's requirements per #197. In other words, the thing doing the configuration should set this to default *when it knows x-ray is likely in use*, so as to not change the default tagging policy for everyone else.

Since the code that is configuring this policy is spring boot 3 (not anything in this repo), and we've also documented. I think this issue needs to be closed.
